lib/harness/pkg.ex

```diff
@@ -24,7 +24,12 @@ defmodule Harness.Pkg do
   """
   @callback links(t()) :: [String.t()]
 
-  defstruct [:path, :module, :config, :files, :name, :links]
+  @doc """
+  A list of hard links to create from the .harness directory to project root.
+  """
+  @callback hard_links(t()) :: [String.t()]
```

Comment:
```suggestion
  @callback hard_links(t()) :: [String.t()]
  @optional_callbacks hard_links: 1, links: 1
```

with default implementations that return the empty list I think will do us well

https://hexdocs.pm/elixir/1.5.2/behaviours.html#optional-callbacks

in the long run, packages should be able to declare which version of the package spec they implement and harness will be able to dispatch the package's configuration to various renderers based on the version declared